### PR TITLE
Move DataRequest::mode_t to ReadoutType enum, use in TriggerDecision and

### PR DESCRIPTION
**Pull request by eflumerf**
DataRequest.

### CODE REVIEW

File: include/dfmessages/Types.hpp [hunk 1]
```diff
@@ -40,6 +42,18 @@ using timestamp_t = dataformats::timestamp_t;           ///< Copy dataformats::t
 using timestamp_diff_t = dataformats::timestamp_diff_t;           ///< Copy dataformats::timestamp_diff_t
 
 using trigger_type_t = dataformats::trigger_type_t; ///< Copy dataformats::trigger_type_t
+
+/**
+ * @brief Which type of readout to use for TriggerDecision and DataRequest
+*/
+enum class ReadoutType : uint16_t
```
Comment (mroda88) [inline include/dfmessages/Types.hpp:49]: Whoever maintains trigger decision generator modules might want to comment here if we need ToString/From string operations. 
I'm not one of those people, so I'll defer and approve.

**Review by mroda88 — APPROVED** [review]
Looks good to me.

**Comment by bieryAtFnal** [comment]
I ran a quick test, and that looked good.
One question: it looked like some of the header files in dfmessages might benefit from clang_format.  Do we want to do that now, or wait until another convenient time?

**Review by floriangroetschla — APPROVED** [review]
Looks good to me too

**Pull request merged**